### PR TITLE
Upgrade ckb dependencies to `v0.111.0`

### PR DESCRIPTION
**Pull request by eval-exec**
- [x] upgrade `ckb-...` dependencies to `0.111.0`
- [x] upgrade unit test

Request for review: @quake

### CODE REVIEW

File: src/test_util.rs [hunk 1]
```diff
@@ -412,6 +428,15 @@ impl TransactionDependencyProvider for Context {
...
         ))
     }
+
+    fn get_block_extension(
+        &self,
+        _block_hash: &Byte32,
+    ) -> Result<Option<ckb_types::packed::Bytes>, TransactionDependencyError> {
+        Err(TransactionDependencyError::NotFound(
+            "block extension not found".to_string(),
+        ))
```
Comment (blckngm) [inline src/test_util.rs:438]: What's the difference between returning `Ok(None)` and an error?

Comment (eval-exec) [reply src/test_util.rs:438]: I unintentionally overlooked the impact of returning an error in this code, and I simply want to ensure that the code compiles without any errors.

Do you think we should return `Ok(None)` here?

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "ckb-sdk"
-version = "2.5.0"
+version = "2.6.0"
```
Comment (blckngm) [inline Cargo.toml:3]: Should be `3.0` because the lifetime change would be breaking.

File: Cargo.toml [hunk 3]
```diff
@@ -29,22 +29,22 @@ lru = "0.7.1"
...
+ckb-script = "0.111.0-rc8"
 bitflags = "1.3.2"
 sha3 = "0.10.1"
 enum-repr-derive = "0.2.0"
 
 # for feature test
 rand = { version = "0.7.3", optional = true }
-ckb-mock-tx-types = { version = "0.108.0", optional = true }
-ckb-chain-spec = { version = "0.108.0", optional = true }
+ckb-mock-tx-types = { git = "https://github.com/nervosnetwork/ckb-standalone-debugger", branch = "develop"}
```
Comment (blckngm) [inline Cargo.toml:46]: If we are going to publish ckb-sdk to crates.io, we would still need to publish this on crates.io.

Comment (eval-exec) [reply Cargo.toml:46]: Yes.

File: src/test_util.rs (hunk 1 above)
Comment (blckngm) [reply src/test_util.rs:438]: I'm getting it now. Error should be returned if the block is not found, and None if the extension is not found, right?

File: src/test_util.rs [hunk 4]
```diff
@@ -354,8 +357,21 @@ impl Context {
...
-
-        let mut verifier = TransactionScriptsVerifier::new(&rtx, &resource);
+        let consensus = ConsensusBuilder::default()
+            .hardfork_switch(HardForks {
+                ckb2021: CKB2021::new_dev_default(),
+                ckb2023: CKB2023::new_dev_default(),
+            })
+            .build();
+        let tip = HeaderBuilder::default().number(0.pack()).build();
+        let tx_verify_env = TxVerifyEnv::new_submit(&tip);
```
Comment (blckngm) [inline src/test_util.rs:367]: Is this the recommended way to build a `TxVerifyEnv` to verify txs offchain?

File: src/test_util.rs (hunk 1 above)
Comment (eval-exec) [reply src/test_util.rs:438]: Yes. 

And the code you commented is used by unit test.

In non-test code, `get_block_extension` has been implemented:

https://github.com/nervosnetwork/ckb-sdk-rust/blob/8bd9d3c6c84fdf901fe2f92f5e499f9a88a51fe2/src/traits/mod.rs#L162-L169

File: src/test_util.rs (hunk 4 above)
Comment (eval-exec) [reply src/test_util.rs:367]: No, this is only a test case `tip` should be the latest tip header, and `TxVerifyEnv` have `new_submit`,`new_commit` methods to get `Self`.

Comment (eval-exec) [reply src/test_util.rs:367]: You can refer to the implementation of the CKB's `estimate_cycles` RPC.

https://github.com/nervosnetwork/ckb/blob/e5b74d01b654a1a4b6cd74d28e19b0b92d630e0e/rpc/src/module/chain.rs#L2354

Comment (blckngm) [reply src/test_util.rs:367]: OK. Thanks.

**Comment by eval-exec** [comment]
I have rebased this branch on master.
Waiting on CI jobs results.

**Comment by eval-exec** [comment]
@quake I think this PR are ready to merge, then we can publish ckb-sdk 3.0.0 now.

**Pull request merged**